### ISSUE TITLE
FGSM attack not working well (cannot replicate adversarial_training_mnist.ipynb under notebooks)

### ISSUE THREAD

**Issue by joelma1** [Trusted-AI/adversarial-robustness-toolbox#670]
When I run the exact script from adversarial_training_mnist.ipynb under notebooks, I get higher model accuracy on adversarial samples than expected. 

![image](https://user-images.githubusercontent.com/64549018/96284218-48cab300-0fab-11eb-92d1-16c5f634769a.png)

For FGSM attack on MNIST test set, the notebook indicated that only 21 images should be correctly classified by the original model, but when I ran the code (using same model/script), I get that 95 images are correctly classified and only 5 adversarial images are misclassified by original model 

![image](https://user-images.githubusercontent.com/64549018/96285066-72d0a500-0fac-11eb-94cb-3298a0aa6455.png)

Furthermore, I've been applying FGSM attacks to the MNIST dataset and model accuracy on adversarial datasets are much higher than expected (around 60% for eps=0.8). 

 - Python 2.7
 - TensorFlow 1.13.1

**Comment by beat-buesser** [comment]
Hiu @joelma1 Thank you very much for raising this issue! Which version of ART are you using?

**Comment by joelma1** [comment]
Version 1.4.0

Thanks!

**Comment by mathsinn** [comment]
@joelma1 This discrepancy is most likely due to a difference in keras and tensorflow versions:

- With keras 2.4.3 and tf 2.3.1, we obtain the same results that you reported (correctly classified: 95, incorrectly: 5)
- With keras 2.3.1 and tf 1.15.0, we obtain correctly classified: 22, incorrectly: 78

The latter are the results we had originally reported in the notebook. In June, after the 1.3 release, we had obtained 21/79, but it appears this was just due to intermittent numerical issues.

We are planning to add more information to the notebook about the exact keras/tf versions that are required for reproducing the exact results.
